zigpy/types/basic.py

```diff
@@ -171,7 +177,9 @@ def serialize(self) -> bytes:
         if self._bits % 8 != 0:
             raise TypeError(f"Integer type with {self._bits} bits is not byte aligned")
 
-        return self.to_bytes(self._bits // 8, "little", signed=self._signed)
+        return self.to_bytes(
+            self._bits // 8, "big" if self._bigendian else "little", signed=self._signed
```

Comment:
And then `"big" if self._bigendian else "little"` would just become `byteorder=self._byteorder`.